Lib/unittest/test/test_case.py

```diff
@@ -835,6 +839,36 @@ def testAssertSequenceEqualMaxDiff(self):
         self.assertGreater(len(msg), len(diff))
         self.assertNotIn(omitted, msg)
 
+    def testAssertEqualModeratelyLongSequencePerformance(self):
+        """Before fixing bpo-19217, assertTrue on different sequences of the
+        same length would never finish in circumstances like the ones below."""
+
+        TIME_LIMIT = 1
+
+        # test comparison of strings
+        arr1 = 'a' * 10000
+        arr2 = 'a' * 9999 + 'b'
+        start = time.time()
+        try:
+            self.assertEqual(arr1, arr2)
+        except self.failureException:
+            pass
+        finish = time.time()
+        if finish - start > TIME_LIMIT:
+            self.fail('String comparison took longer than one second')
+
+        # test comparison of lists
+        arr1 =  [1] * 10000
+        arr2 = ([1] * 9999) + [2]
+        start = time.time()
```

Comment:
This is again a non-representative example, taking the same time before and after the change. You need to generate a diff with more than 10 differences to see the difference in timing.

For example, if I use the following, there's a big difference:
```py3
        arr1 =  [1] * 1000
        a2 = ([1] * 99) + [2]
        arr2 = a2 * 10
```

New version takes 0.05s whereas the original takes a whopping 150s **and** requires raising the recursion limit for it to show the final diff 😂. And the increase is exponential, if you change the list size to 2,000 then the original takes a whopping 822s!

Sadly, [the old version looked quite good](https://gist.github.com/ambv/da471cb6e45beafa0068f4e5ceda00e8) while [the new one is pretty useless](https://gist.github.com/ambv/880e6cff441250d422e17f5fbf93c217). But that differs between cases so I guess it's better to get a bad diff than no diff in 15 minutes.

Reply:
I didn't think testing runtime was a good idea either, but I was hoping for an answer via Cunningham's Law...

Speaking of the recursion limit, maybe that's a way to get a better regression test? 

```python
height = len(inspect.stack(0))
original_limit = sys.getrecursionlimit()
sys.setrecursionlimit(height + <whatever we need>)
# do test
# restore recursion limit
```
What do you think?

Reply:
Nah, that idea is not good because as you said, the recursion only happens when the result is printed, not as it's being generated.

Reply:
Yeah, we don't need to touch the recursion limit just for this test. In fact, the passing version won't be exceeding the recursion limit anyway.